vertical-pod-autoscaler/pkg/updater/main.go

```diff
@@ -113,9 +113,9 @@ func main() {
 		klog.Fatalf("Failed to create updater: %v", err)
 	}
 	ticker := time.Tick(*updaterInterval)
-	ctx, cancel := context.WithTimeout(context.Background(), *updaterInterval)
-	defer cancel()
 	for range ticker {
+		ctx, cancel := context.WithTimeout(context.Background(), *updaterInterval)
+		defer cancel()
```

Comment:
What is defer vs for interaction. Are defers run at the end of a for iteration or after exiting a for?

Reply:
They're run when the function returns. So here we would never run the defer. Thanks for catching the bug!

I'll check if we have similar bug repeated elsewhere (using defer in main / possible a different function from which we don't return)